### PR TITLE
Enhance Admin CP Plugins List

### PR DESCRIPTION
**Pull request by JordanMussi**
- Converts links to buttons
- Adds configure button (link set by the `_configure` plugin function) shown when plugin is installed and returns a configure link
- Documentation added to `hello` plugin

### CODE REVIEW

File: admin/modules/config/plugins.php [hunk 1]
```diff
@@ -663,31 +665,47 @@ function build_plugin_list($plugin_list)
...
 			{
-				$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=activate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\">{$lang->install_and_activate}</a>", array("class" => "align_center", "colspan" => 2));
+				$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=activate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\" class=\"button\">{$lang->install_and_activate}</a>", array("class" => "align_center", "colspan" => 3));
 			}
 		}
 		// Plugin is activated and installed
 		else if($plugininfo['is_active'])
 		{
-			$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=deactivate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\">{$lang->deactivate}</a>", array("class" => "align_center", "width" => 150));
+			if ($configure_link !== false) {
```
Comment (euantorano) [inline admin/modules/config/plugins.php]: Code style - this should have no space between `if` and the `(` and the `{` should be on a new line.

Comment (JordanMussi) [reply admin/modules/config/plugins.php]: Done in e3d47839eb1e6e066bdddc614b24e27f90f7816b 👻

**Comment by Ben-MyBB** [comment]
@euantorano I assume this is something we would be looking to add into 1.10?

**Comment by euantorano** [comment]
Probably 1.9.0 I’d say, since the work is already done :)

File: inc/plugins/hello.php [hunk 2]
```diff
@@ -83,6 +83,27 @@ function hello_info()
...
+{
+	global $db;
+
+	$query = $db->simple_select('settinggroups', 'gid', "name='hello'");
+
+	$gid = (int)$db->fetch_field($query, 'gid');
+
+	if($gid > 0)
+	{
+		return "index.php?module=config-settings&amp;action=change&amp;gid={$gid}";
```
Comment (Matslom) [inline inc/plugins/hello.php]: Is returning the url the best option? Future routing improvements may change url structure. 
IMHO returned `GID` gives us more freedom.

Comment (euantorano) [reply inc/plugins/hello.php]: I'd agree with that @Matslom. Returning GID seems much more useful to me.

Comment (JordanMussi) [reply inc/plugins/hello.php]: I went with URL as many plugins use a new module to hold their configuration. Only accepting a GID as the return would prevent the plugin from linking to the module.

**Comment by Ben-MyBB** [comment]
I personally think it would be good to get this in 1.9, ping @JordanMussi

**Comment by JordanMussi** [comment]
@Ben-MyBB I've rebased the changes onto 1.9 https://github.com/mybb/mybb/compare/develop/1.9...JordanMussi:enhance/acp-plugins-list. I can't seem to reopen this PR to update the changes maybe that's something for a maintainer to do? Not too sure why it hasn't shown the updates to the branch here as well (should be showing force pushed to 2f8bb4cce5437bb51cd32ab3349550cacaae043d)

**Comment by euantorano** [comment]
I can’t re-open it either for whatever weird reason - the button is disabled for me too.

**Comment by Ben-MyBB** [comment]
When we hover over the re-open button it says that we can't reopen since it has been force pushed or recreated?

**Comment by JordanMussi** [comment]
If you’re happy with the changes can just create a new PR I suppose :-)

**Comment by euantorano** [comment]
Yeah, I'd like to see a new PR for sure. The only issue I originally had was returning a link to the settings, but that's not. Abigails deal and we can change it in the future when we make ACP changes.

**Pull request closed without merging**